src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-list/cephfs-list.component.ts

```diff
@@ -87,4 +127,56 @@ export class CephfsListComponent extends ListWithDetails implements OnInit {
   updateSelection(selection: CdTableSelection) {
     this.selection = selection;
   }
+
+  removeVolumeModal() {
+    const volName = this.selection.first().mdsmap['fs_name'];
+    console.log("waah bc", volName);
+    this.modalService.show(CriticalConfirmationModalComponent, {
+      itemDescription: 'Volume',
+      itemNames: [volName],
+      submitActionObservable: () =>
+        this.taskWrapper.wrapTaskAroundCall({
+          task: new FinishedTask('cephfs/remove', { name: volName }),
+          call: this.cephfsService.remove(volName)
+        })
+    });
+  }
+
+  getDisableDesc(): boolean | string {
+    if (this.selection?.hasSelection) {
+      if (!this.monAllowPoolDelete) {
+        return $localize`Volume deletion is disabled by the mon_allow_pool_delete configuration setting.`;
+      }
+
+      return false;
+    }
+
+    return true;
+  }
+
+  editAction() {
+    const selectedVolume = this.selection.first().mdsmap['fs_name'];
+
+    this.modalService.show(FormModalComponent, {
+      titleText: $localize`Edit Volume: ${selectedVolume}`,
+      fields: [
+        {
+          type: 'text',
+          name: 'volumeName',
+          value: selectedVolume,
+          label: $localize`Name`,
+          required: true
+        }
+      ],
```

Comment:
I just noticed this. We already have a form component for the volume creation. Why not reuse that and disable whatever that is not needed for the edit? Otherwise whenever we add new things to the creation form, we'll need to explicitly maintain the edit form as well. So we can follow the similar pattern that is done all over the dashboard for this edit form as well.